contracts/seed/Seed.sol

```diff
@@ -0,0 +1,376 @@
+      * @param _fee                   Success fee as %
+    */
+    function initialize(
+            address _beneficiary,
+            address _admin,
+            address _seedToken,
+            address _fundingToken,
+            uint[] memory    _successMinimumAndCap,
+            uint    _price,
+            uint    _startTime,
+            uint    _endTime,
+            uint16  _vestingDuration,
+            uint16  _vestingCliff,
+            bool    _isWhitelisted,
+            uint8   _fee
+    ) public initializer {
+        beneficiary     = _beneficiary;
+        admin           = _admin;
+        successMinimum  = _successMinimumAndCap[0];
+        cap             = _successMinimumAndCap[1];
+        price           = _price;
+        startTime       = _startTime;
+        endTime         = _endTime;
+        vestingDuration = _vestingDuration;
+        vestingCliff    = _vestingCliff;
+        isWhitelisted   = _isWhitelisted;
+        seedToken       = IERC20(_seedToken);
+        fundingToken    = IERC20(_fundingToken);
+        fee             = _fee;
+        closed          = false;
+        minimumReached  = false;
+    }
+
+    /**
+      * @dev                     Buy seed tokens.
+      * @param _amount           The amount of tokens to buy.
+    */
+    function buy(uint256 _amount) public protected checked {
+        require((fundingToken.balanceOf(address(this)) + _amount) <= cap, "Seed: amount exceeds contract sale cap");
+        require(fundingToken.transferFrom(msg.sender, address(this), _amount), "Seed: no tokens");
+
+        if (fundingToken.balanceOf(address(this)) >= successMinimum) {
+            minimumReached = true;
+        }
+
+        fundingTokensPerAddress[msg.sender] = _amount;
+
+        uint feeAmount = (_amount.mul(uint(PPM))).mul(fee).div(100000000); // 100000000 = PPM * 100
+        fees[msg.sender] = feeAmount;
+
+        uint _lockTokens = tokenLocks[msg.sender].amount;
+        _addLock(msg.sender, block.timestamp, (_lockTokens.add(_amount)).mul(price).div(PCT_BASE));
+    }
+
+    /**
+      * @dev                     Claim locked tokens.
+      * @param _locker           The address of the locker.
+    */
+    function claimLock(address _locker) public checkMinimumReached {
+        uint16 daysVested;
+        uint256 amountVested;
+        (daysVested, amountVested) = _calculateClaim(_locker);
+        require(amountVested > 0, "Seed: amountVested is 0");
+
+        Lock storage tokenLock = tokenLocks[_locker];
+        tokenLock.daysClaimed = uint16(tokenLock.daysClaimed.add(daysVested));
+        tokenLock.totalClaimed = uint256(tokenLock.totalClaimed.add(amountVested));
+
+        require(seedToken.transfer(beneficiary, fees[_locker]), "Seed: cannot transfer to beneficiary");
+        require(seedToken.transfer(tokenLock.recipient, amountVested), "Seed: no tokens");
+        emit TokensClaimed(tokenLock.recipient, amountVested);
+    }
+
+    // returns funding tokens to user
```

Comment:
/**
      * @dev                            Returns funding tokens to user.
    */